packages/repository/src/__tests__/unit/repositories/legacy-juggler-bridge.unit.ts

```diff
@@ -416,4 +418,21 @@ describe('DefaultCrudRepository', () => {
       'execute() must be implemented by the connector',
     );
   });
+
+  it('throws an error when beginTransaction() not implemented', async () => {
+    const repo = new DefaultCrudRepository(Note, ds);
+    await expect(repo.beginTransaction({})).to.be.rejectedWith(
+      'beginTransaction must be function implemented by the connector',
+    );
+  });
+  it('calls connector beginTransaction() when available', async () => {
+    const crudDs = new juggler.DataSource({
+      name: 'db',
+      connector: CrudConnectorStub,
+    });
+
+    const repo = new DefaultCrudRepository(Note, crudDs);
+    const res = await repo.beginTransaction({});
```

Comment:
If we are accepting an empty array then let's make the argument optional please.

```suggestion
    const res = await repo.beginTransaction();
```